main/main.go

```diff
@@ -234,27 +234,67 @@ func main() {
 			)
 			panic(err)
 		}
-		subnetInfo := s
+		sourceBlockchain := s
 
 		health := atomic.NewBool(true)
 		relayerHealth[blockchainID] = health
 
 		// errgroup will cancel the context when the first goroutine returns an error
 		errGroup.Go(func() error {
-			// runListener runs until it errors or the context is cancelled by another goroutine
-			return runListener(
+			// Dial the eth client
+			ethClient, err := ethclient.DialWithConfig(
+				context.Background(),
+				sourceBlockchain.RPCEndpoint.BaseURL,
+				sourceBlockchain.RPCEndpoint.HTTPHeaders,
+				sourceBlockchain.RPCEndpoint.QueryParams,
+			)
+			if err != nil {
+				logger.Error(
+					"Failed to connect to node via RPC",
+					zap.String("blockchainID", sourceBlockchain.BlockchainID),
+					zap.Error(err),
+				)
+				return err
+			}
+
+			// Create the ApplicationRelayers
+			applicationRelayers, minHeight, err := createApplicationRelayers(
```

Comment:
Should we move the creation of the `ethClient` and `applicationRelayers` out of the anonymous function run use `errGroup.Go`? I know we don't need references to them elsewhere right now, but could be useful in the future for routing messages to the application relayers otherwise.

Reply:
The thought process is that they're easy to hoist out of the goroutine if we need to, but right now processing them concurrently will save a bit of time.